src/main/java/apoc/load/LoadXls.java

```diff
@@ -172,17 +180,60 @@ private Object convertArray(Object value) {
 
         private Object convertType(Object value) {
             if (nullValues.contains(value) || value == null) return null;
-//            if (!(value instanceof String)) return value;
             switch (type) {
-                case STRING: return value.toString();
-                case INTEGER: return Util.toLong(value);
-                case FLOAT: return Util.toDouble(value);
-                case BOOLEAN: return Util.toBoolean(value);
-                case NULL: return null;
-                case LIST: return Arrays.stream(arrayPattern.split(value.toString())).map(this::convertType).collect(Collectors.toList());
-                default: return value;
+                case STRING:
+                    if (value instanceof Temporal && dateFormat instanceof String && dateFormat != null) {
+                        return DateTimeFormatter.ofPattern(dateFormat.toString()).format((TemporalAccessor) value);
+                    } else {
+                        return value.toString();
+                    }
+                case INTEGER:
+                    return Util.toLong(value);
+                case FLOAT:
+                    return Util.toDouble(value);
+                case BOOLEAN:
+                    return Util.toBoolean(value);
+                case NULL:
+                    return null;
+                case LIST:
+                    return Arrays.stream(arrayPattern.split(value.toString())).map(this::convertType).collect(Collectors.toList());
+                case DATE:
+                    return temporalFormat(value, LocalDate.class);
+                case DATE_TIME: case LOCAL_DATE_TIME:
+                    return temporalFormat(value, LocalDateTime.class);
+                case LOCAL_TIME: case TIME:
+                    return temporalFormat(value, LocalTime.class);
+                default:
+                    return value;
             }
         }
+
+        private Object temporalFormat(Object value, Class<?> temporalClass) {
+            if (value instanceof String) {
+                try {
+                    switch (temporalClass.getSimpleName()){
```

Comment:
Non è buona cosa affidarsi al nome delle classi, se non stettamente necessiario, in questo caso è meglio:
```suggestion
                    if (LocalDate.class.equals(temporalClass))
```

Reply:
Se la stringa contiene il timezone cosa succede?
Fare un esempio

Reply:
Come viene gestita questa `2011-01-01T12:00:00.05381+05:00`?